tests/qos/qos_sai_base.py

```diff
@@ -596,77 +743,126 @@ def dutConfig(
                             uplinkPortIps.append(portConfig["peer_addr"])
                             uplinkPortNames.append(intf)
 
-            testPortIps = self.__assignTestPortIps(mgFacts, topo)
+            testPortIps[src_dut_index] = {}
+            testPortIps[src_dut_index][src_asic_index] = self.__assignTestPortIps(src_mgFacts)
```

Comment:
data model changes of testPortIds and testPortIps will break some functions. please help to resolve them as well.

Reply:
The tests were adjusted to work for broadcom chipset, so missed the failures reported due to changed data model of testPortIds and testPortsIps. Pushed fixes in latest commit - please test it out.